<?xml version="1.0"?>
<launch>
  <!-- sync gazebo time with ros -->
  <param name="use_sim_time" value="true" />

  <!-- settings -->
  <param name="sim_mode" value="true" />
  <param name="debug_mode" value="false" />

  <!-- prototype -->

  <param name="pose_stamped_mode" value="true" />
  <param name="position_topic" value="/awesomo/gimbal/frame/pose" />
  <param name="orientation_topic" value="/awesomo/gimbal/joint/orientation" />

  <node pkg="prototype_ros" name="prototype_gazebo" type="run_gazebo.sh" output="screen" required="true" />
  <node pkg="prototype_ros" name="prototype_world" type="world_node" output="screen" required="true" />
  <node pkg="prototype_ros" name="prototype_quadrotor" type="quadrotor_node" output="screen" required="true" />
  <node pkg="prototype_ros" name="prototype_camera" type="camera_node" output="screen" required="true">
    <remap from="/prototype/camera/image" to="/awesomo/camera/image" />
    <remap from="/prototype/camera/image_pose_stamped" to="/awesomo/camera/image_pose_stamped" />
  </node>
  <node pkg="prototype_ros" name="prototype_gimbal" type="gimbal_node" output="screen" required="true" >
    <remap from="/prototype/gimbal/frame/pose" to="/awesomo/gimbal/frame/pose" />
    <remap from="/prototype/gimbal/joint/orientation" to="/awesomo/gimbal/joint/orientation" />
    <remap from="/prototype/gimbal/setpoint/attitude" to="/awesomo/gimbal/setpoint/attitude" />
    <remap from="/prototype/gimbal/track" to="/awesomo/gimbal/track" />
    <remap from="/prototype/gimbal/target/local" to="/awesomo/gimbal/target/local" />
    <remap from="/prototype/gimbal/target/world" to="/awesomo/gimbal/target/world" />
  </node>

  <!-- control node -->
  <param name="control_config_dir" value="$(find awesomo_configs)/configs/quadrotor_gazebosim" />
  <node pkg="awesomo_ros" name="awesomo_control" type="control_node" output="screen" />

  <!-- apriltag node -->
  <param name="apriltag_config" type="str" value="$(find awesomo_configs)/configs/apriltag/prototype.yaml" />
  <node pkg="awesomo_ros" name="awesomo_apriltag" type="apriltag_node" output="screen" required="true"/>

  <!-- estimate node -->
  <node pkg="awesomo_ros" name="awesomo_estimator" type="estimator_node" output="screen" required="true"/>
</launch>
